cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -431,6 +431,57 @@ func (s *Server) repoDetail(ctx context.Context, repoRef *corev1.PackageReposito
 	}, nil
 }
 
+func (s *Server) repoSummaries(ctx context.Context, namespace string) ([]*corev1.PackageRepositorySummary, error) {
+	summaries := []*corev1.PackageRepositorySummary{}
+	var repos []sourcev1.HelmRepository
+	var err error
+	if namespace == apiv1.NamespaceAll {
+		if repos, err = s.listReposInAllNamespaces(ctx); err != nil {
+			return nil, err
+		}
+	} else {
+		// here I think the right semantics are different than that of availablePackageSummaries
+		// namely, if a specific namespace is specified we need to list repos in that namespace
+		// and if the caller happens not to have 'read' access to that namespace, a PermissionDenied
+		// error should be raised, as opposed to returning an empty list with no error
+
+		// also note that, at least theoretically, it is not clear what we'd do if we ever
+		// had a flavor of API that allowed 2 namespaces (as opposed to all/one), and the caller
+		// only had access to 1 of them. Raise the error (same as with one namespace) or ignore
+		// inaccessible namespace (same as with all namespaces)?
```

Comment:
Not sure we need this note, but yeah, I'd expect a 404 to b raised in that case (that is, explicitly requesting a namespace, whether it's one of many or alone, that I cannot access should return an explicit error).